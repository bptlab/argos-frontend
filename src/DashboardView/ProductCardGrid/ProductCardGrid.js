import React, { Component } from 'react';
import './ProductCardGrid.css';
import ProductCard from './ProductCard/ProductCard.js';

class ProductCardGrid extends Component {
    searchMatches(product) {
        if (!this.props.searchText) {
            return true;
        }
        //SEARCH-CONFIG: Edit this to define relevant fields for overview search
        const searchFields = [
            product.name,
            product.orderNumber,
            product.productionStart
        ];

        return !searchFields.every((searchField) => {
            return !(searchField.toString().toLowerCase().indexOf(this.props.searchText) > -1);
        });
    }

    render() {
        return (
            <div className="container">
                <div className="row">
                    {this.props.products.map((product) =>  {
                        if(this.searchMatches(product)) {
                            return (
                                <ProductCard
                                    key={product.id}
                                    product={product}/>
                            );
                        } else {
                            return false;
                        }
                    })}
                </div>
            </div>
        );
    }
}
export default ProductCardGrid;