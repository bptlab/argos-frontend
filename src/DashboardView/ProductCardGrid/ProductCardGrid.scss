.ProductCardGrid {
    display: flex;
    align-items: flex-start;
    flex-wrap: wrap;
    justify-content: space-between;
}