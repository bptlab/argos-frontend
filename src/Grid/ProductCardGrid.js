import React, { Component } from 'react';
import './ProductCardGrid.scss';

class Grid extends Component {
    render() {
        return (
            <div className="ProductCardGrid">
                {this.props.children}
            </div>
        );
    }
}

export default Grid;
