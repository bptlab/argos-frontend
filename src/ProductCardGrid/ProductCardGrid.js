import React, { Component } from 'react';
import './ProductCardGrid.css';
import ProductCard from '../ProductCard/ProductCard.js';

class ProductCardGrid extends Component {

    searchMatches(product) {
        if (!this.props.searchText) {
            return true;
        }
        let searchFields = [product.metaData.brand, product.metaData.label, product.metaData.orderNumber,
            product.productionStart];
        for(let index in searchFields) {
            if (searchFields[index].toString().indexOf(this.props.searchText) > -1) {
                return true;
            }
        }
        return false;
    }

    render() {
        return (
            <div className="ProductCardGrid">
                {this.props.products.map((product) =>  {
                    if(this.searchMatches(product)) {
                        return (<ProductCard key={product.id} product={product}/>);
                    } else {
                        return false;
                    }
                })}
            </div>
        );
    }
}

export default ProductCardGrid;
