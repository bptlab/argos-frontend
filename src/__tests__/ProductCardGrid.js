import React from 'react';
import ProductCardGrid from '../DashboardView/ProductCardGrid/ProductCardGrid.js';
import renderer from 'react-test-renderer';
import TestData from './testData/products.js'
let instance;

test("Correct drawing of ProductCardGrid", () => {
    const component = renderer.create(
        <ProductCardGrid
            ref={(child) => {instance = child}}
            products={TestData.PRODUCTS}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Correct search of products", () => {
    let testResult = instance.searchMatches(TestData.PRODUCTS[0], TestData.PRODUCTS[0].name);
    expect(testResult).toEqual(true);
    testResult = instance.searchMatches(TestData.PRODUCTS[0], "Something unreal: #+343");
    expect(testResult).toEqual(false);
});