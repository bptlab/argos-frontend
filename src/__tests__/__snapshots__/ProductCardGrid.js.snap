exports[`test Correct drawing of ProductCardGrid 1`] = `
<div
  className="container">
  <div
    className="row">
    <div
      className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
      <a
        href="/#/product/0"
        title="Details for Testproduct 5000 14E-17B">
        <div
          className="card ProductCard ProductCard_RUNNING">
          <h4
            className="card-title ProductCardTitle ProductCardTitle_RUNNING">
            Testproduct 5000 14E-17B
          </h4>
          <ul>
            <li
              className="ProductCardInformationLine PCIL_serialNumber">
              9830510416
            </li>
            <li
              className="ProductCardInformationLine PCIL_quantity">
              52189
            </li>
            <li
              className="ProductCardInformationLine PCIL_date">
              2016-05-07
            </li>
            <li
              className="ProductCardInformationLine PCIL_errorQuantity">
              84
            </li>
          </ul>
        </div>
      </a>
    </div>
    <div
      className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
      <a
        href="/#/product/1"
        title="Details for Testproduct 5100 12E-15B">
        <div
          className="card ProductCard ProductCard_WARNING">
          <h4
            className="card-title ProductCardTitle ProductCardTitle_WARNING">
            Testproduct 5100 12E-15B
          </h4>
          <ul>
            <li
              className="ProductCardInformationLine PCIL_serialNumber">
              7840516417
            </li>
            <li
              className="ProductCardInformationLine PCIL_quantity">
              3215
            </li>
            <li
              className="ProductCardInformationLine PCIL_date">
              2016-03-01
            </li>
            <li
              className="ProductCardInformationLine PCIL_errorQuantity">
              54
            </li>
          </ul>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`test Correct drawing of ProductCardGrid in case of excluded state 1`] = `
<div
  className="container">
  <div
    className="row">
    <div
      className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
      <a
        href="/#/product/0"
        title="Details for Testproduct 5000 14E-17B">
        <div
          className="card ProductCard ProductCard_RUNNING">
          <h4
            className="card-title ProductCardTitle ProductCardTitle_RUNNING">
            Testproduct 5000 14E-17B
          </h4>
          <ul>
            <li
              className="ProductCardInformationLine PCIL_serialNumber">
              9830510416
            </li>
            <li
              className="ProductCardInformationLine PCIL_quantity">
              52189
            </li>
            <li
              className="ProductCardInformationLine PCIL_date">
              2016-05-07
            </li>
            <li
              className="ProductCardInformationLine PCIL_errorQuantity">
              84
            </li>
          </ul>
        </div>
      </a>
    </div>
  </div>
</div>
`;
